unipop-elastic/src/org/unipop/elastic/document/DocumentController.java

```diff
@@ -135,48 +178,69 @@ public Vertex addVertex(AddVertexQuery uniQuery) {
     //endregion
 
     //region Elastic Queries
-    private <E extends Element, S extends DocSchema<E>> Iterator<E> search(PredicatesHolder allPredicates, Set<S> schemas, int limit, StepDescriptor stepDescriptor) {
-        if(schemas.size() == 0 || allPredicates.isAborted()) return Iterators.emptyIterator();
+    private <E extends Element, S extends DocSchema<E>> Iterator<E> search(PredicatesHolder allPredicates, Set<S> schemas, int limit) throws IOException {
+        if(schemas.size() == 0 || allPredicates.isAborted()) return Collections.emptyIterator();
 
         FilterBuilder filterBuilder = FilterHelper.createFilterBuilder(allPredicates);
         QueryBuilder query = QueryBuilders.filteredQuery(QueryBuilders.matchAllQuery(), filterBuilder);
 
         String[] indices = schemas.stream().map(DocSchema::getIndex).toArray(String[]::new);
         refresh(indices);
-        QueryIterator.Parser<E> parser = (searchHit) -> schemas.stream().map(schema -> schema.fromFields(searchHit.getSource())).findFirst().get();
-        return new QueryIterator<>(query, stepDescriptor, 0, limit, client, parser, indices);
+        QueryIterator.Parser<E> parser = (searchHit) -> schemas.stream().map(schema -> schema.fromFields(searchHit)).findFirst().get();
+        return new QueryIterator<>(query, 0, limit, client, parser, indices);
     }
 
-    private <E extends Element> IndexResponse index(Set<? extends DocSchema<E>> schemas, E element, boolean create) {
+    private <E extends Element> DocumentResult index(Set<? extends DocSchema<E>> schemas, E element, boolean create) throws IOException {
         for(DocSchema<E> schema : schemas) {
             DocSchema.Document document = schema.toDocument(element);
-            if (document!= null) {
-                IndexRequestBuilder indexRequest = client.prepareIndex(document.getIndex(), document.getType(), document.getId())
-                        .setSource(document.getFields()).setCreate(create);
-                dirty = true;
-                return indexRequest.execute().actionGet();
+            if (document != null) {
+                if (create) {
+                    Get getRequest = new Get.Builder(document.getIndex(), document.getId()).build();
```

Comment:
It does not suprisingly. atleast not through the `Index.Builder` class. 
an option is to use the default `elasticsearch API` and execute it through `Jest`.